src/hotspot/share/runtime/sweeper.cpp

```diff
@@ -442,6 +437,15 @@ void NMethodSweeper::sweep_code_cache() {
     CompileBroker::set_should_compile_new_jobs(CompileBroker::run_compilation);
     log.debug("restart compiler");
     log_sweep("restart_compiler");
+    EventJitRestart event;
```

Comment:
It would be good to have a sanity test of the event. 

If the event can't be provoked reliably, retry or accept as OK.